src/app/lists/dataview.tsx

```diff
@@ -0,0 +1,199 @@
+"use client";
+
+import Table, { Datum } from "@/components/Tables/Table";
+import React, { useState } from "react";
+import { styled } from "styled-components";
+import EditModal from "@/app/lists/editModal";
+import supabase, { Schema } from "@/supabase";
+import ConfirmDialog from "@/components/Modal/Confirm";
+import Snackbar from "@mui/material/Snackbar/Snackbar";
+import Alert from "@mui/material/Alert/Alert";
+
+export type ListRow = Schema["lists"];
+
+type Props = {
+    data: ListRow[] | null;
+};
+
+export const headers: [string, string][] = [
+    ["item_name", "Description"],
+    ["1_quantity", "Single"],
+    ["2_quantity", "Family of 2"],
+    ["3_quantity", "Family of 3"],
+    ["4_quantity", "Family of 4"],
+    ["5_quantity", "Family of 5"],
+    ["6_quantity", "Family of 6"],
+    ["7_quantity", "Family of 7"],
+    ["8_quantity", "Family of 8"],
+    ["9_quantity", "Family of 9"],
+    ["10_quantity", "Family of 10+"],
+];
+
+export const tooltips: [string, string][] = [
+    ["1_notes", "Single"],
+    ["2_notes", "Family of 2"],
+    ["3_notes", "Family of 3"],
+    ["4_notes", "Family of 4"],
+    ["5_notes", "Family of 5"],
+    ["6_notes", "Family of 6"],
+    ["7_notes", "Family of 7"],
+    ["8_notes", "Family of 8"],
+    ["9_notes", "Family of 9"],
+    ["10_notes", "Family of 10+"],
+];
+
+const ListsDataView: React.FC<Props> = ({ data: rawData }) => {
+    // null => add, undefined => modal closed
+    const [modal, setModal] = useState<Datum | null | undefined>(undefined);
+    const [toDelete, setToDelete] = useState<number | null>(null);
+    // need another setState otherwise the modal content changes before the close animation finishes
+    const [toDeleteModalOpen, setToDeleteModalOpen] = useState<boolean>(false);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    if (rawData === null) {
+        throw new Error("No data found");
+    }
+
+    const remapTooltips = (
+        row: ListRow
+    ): Datum & { unmappedTooltips: { [key: string]: string } } => {
```

Comment:
Can you extract these all to interfaces? (potentially using 'extends')
`{[key: string]: string}` is enough on one line to be a separate interface, in my opinion.

Reply:
Fixed, turns out there was one but I wasn't using it